mirascope/base/types.py

```diff
@@ -1,11 +1,46 @@
 """Types for working with Mirascope prompts."""
-from typing import Callable, Iterable, Literal, Optional, Type, TypedDict, Union
+from enum import Enum
+from inspect import isclass
+from typing import (
+    Annotated,
+    Any,
+    Callable,
+    Iterable,
+    Literal,
+    Optional,
+    Type,
+    TypedDict,
+    Union,
+    get_origin,
+)
 
 from pydantic import BaseModel
 from typing_extensions import Required
 
 from .tools import BaseTool
 
+BaseType = Union[
+    str,
+    int,
+    float,
+    bool,
+    list,
+    dict,
```

Comment:
extracting a dict gives Pydantic validation error:
pydantic_core._pydantic_core.ValidationError: 1 validation error for DictTool
value
  Field required [type=missing, input_value={'tool_call': ChatComplet...ool'), type='function')}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.6/v/missing

Reply:
Ok after further testing and research, it appears that it just doesn't work with dict, so I've removed support for dict as a base class (the tool just isn't populated in this case).

Will punt to further research.